### PR TITLE
Fix setting  multiple interval error

### PR DESCRIPTION
**Pull request by beizhedenglong**

### CODE REVIEW

**Comment by renatorib** [comment]
I do not know if I understood the idea here. What is the problem?

**Comment by beizhedenglong** [comment]
> I do not know if I understood the idea here. What is the problem?

@renatorib if we call start multiple times. We can only stop the last interval. And the previous intervals still running, we can’t stop them

**Comment by renatorib** [comment]
Oh, I got it.
So, I think clearInterfal before set a new is better, what you think?

```jsx
  _setIntervalIfNecessary = delay => {
    if (Number.isFinite(delay)) {
      this._clearIntervalIfNecessary()
      this.intervalId = setInterval(
        () => this.setState(s => ({ times: s.times + 1 })),
        delay
      )
    }
  }
```

**Comment by beizhedenglong** [comment]
It's ok.  But ...
```js
_setIntervalIfNecessary = delay => {
  if (Number.isFinite(delay)) {
    this._clearIntervalIfNecessary()
    //  If we already have an interval ` a`, this will create a new interval has the same functionality  that `a` has.
    // Because We always remove old interval and create a new one in componentDidUpdate phrase, when delay changes.
    this.intervalId = setInterval(
      () => this.setState(s => ({ times: s.times + 1 })),
      delay
    )
  }
}

componentDidUpdate(prevProps) {
  // already removed old interval here.
  if (prevProps.delay !== this.props.delay) {
    this.stop()
    this.start()
  }
}

```js

I think maybe just ignore the old interval is better.

**Comment by renatorib** [comment]
I think I prefer keep `start` working as 'restart'. Powerplug was born to be practical, to be uncomplicated. I think it's better.

**Comment by renatorib** [comment]
I think I prefer keep start working as 'restart'. Powerplug was born to be practical, to be uncomplicated. I think it's better.

**Comment by beizhedenglong** [comment]
it's fine to restart interval when calling  `start`

File: src/components/Interval.js [hunk 1]
```diff
@@ -15,7 +15,7 @@ class Interval extends Component {
   }
 
   _setIntervalIfNecessary = delay => {
-    if (Number.isFinite(delay)) {
+    if (Number.isFinite(delay) && !this.intervalId) {
```
Comment (renatorib) [inline src/components/Interval.js]: I think removing the check `&& !this.intervalId` and adding `this._clearIntervalIfNecessary()` right after solve the problem. Can you do that so we can merge PR?

**Comment by renatorib** [comment]
Nice! I requested this change in code review so we can continue, okay? Thanks!

File: src/components/Interval.js (hunk 1 above)
Comment (beizhedenglong) [reply src/components/Interval.js]: done~

**Comment by renatorib** [comment]
Thanks @beizhedenglong

**Pull request merged**